### ISSUE TITLE
Seafile integration

### ISSUE THREAD

**Issue by reeseovine** [LibrePhotos/librephotos#247]
**Describe the enhancement you'd like**
I would like to be able to scan photos in a [Seafile](https://github.com/haiwen/seafile) library, much like you can with Nextcloud.

**Describe why this will benefit the LibrePhotos**
This would expand the options for image sources for users who already have them in Seafile and don't want to switch to Nextcloud or an unencrypted local storage solution.

**Additional context**
Here is the documentation for Seafile's web API: https://download.seafile.com/published/web-api/home.md. I would be able to help with testing commits but I'm not sure if I'll have the time or energy to dig through the LP codebase and write a proper solution myself.

**Comment by tomamplius** [comment]
nextcloud, seafile etc... maybe be is usefull will be use webdavfs for all this thirdparty apps?

**Comment by derneuere** [comment]
Yes webdav is definitly also a good idea, but seafile is special because it saves all files in a database and not in a filesystem.

**Comment by tomamplius** [comment]
I think is not an issue if app provide webdav fs gateway
https://manual.seafile.com/extension/webdav/

**Comment by reeseovine** [comment]
WebDAV would be fine too!

**Comment by derneuere** [comment]
@tomamplius Ah cool. Nextcloud also supports webdavfs, I think it would make sense then to switch to webdavfs and replace the old scanning. I think we don't even have to download all the files into the local folder then :)

**Comment by tomamplius** [comment]
yes is native on onwcloud and nextcloud.
for next cloud
~~~
apt install davfs2
mount.davfs https://<serveur url>/remote.php/dav/files/<user>/ <mount point>
~~~
But i think that maybe the lock file will not work (like old nfs). I need to test on libresphoto to confirm that
~~~
mount.davfs: warning: the server does not support locks

**Comment by tomamplius** [comment]
It's work!!!!!!

Add url login and password in /etc/davfs2/secrets
exemple :
~~~
https://monnextcloud.mondomaine.fr/remote.php/dav/files/myuser myuser password
~~~
in /etc/fstab 
~~~
https://monnextcloud.mondomaine.fr/remote.php/dav/files/myuser  /var/lib/librephotos/data/photos davfs uid=librephotos,gid=librephotos,file_mode=0777,dir_mode=0777,_netdev 0 0
~~~
